pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -536,6 +533,7 @@ public static void persistCreationMeta(File indexDir, long crc, long creationTim
    */
   void buildIndexCreationInfo()
       throws Exception {
+    long statsCollectorStartTime = System.nanoTime();
```

Comment:
This part is actually not the stats collection. Stats collection happens within `init()`: `dataSource.gatherStats()`. We should be able to move that into `build()`

Reply:
Oh okay, so should the total stats collection time be simply the time taken for `dataSource.gatherStats()` or should it be summed with the time taken for `buildIndexCreationInfo()`? Also I think it might be better to leave it in `init` rather than moving it into `build` since the segment stats are also used in `buildByColumn`?

Reply:
I'd suggest combining `dataSource.gatherStats()` and `buildIndexCreationInfo()` into one step, and count it as the stats collection in `build()` or `buildByColumn()` method. Currently `init()` can be very heavy because of the stats collection, and the time spent there is not even logged

Reply:
Makes sense, I've verified that all usages of `_segmentStats` / `_totalDocs` are only after `build()` / `buildByColumn()` anyway so there's no need to have it in `init()` 👍